onboarding_tool/obtmain.c

```diff
@@ -1647,12 +1646,19 @@ provision_ace2(void)
     return;
   }
 
-  while (num_resources <= 0 || num_resources > MAX_NUM_RESOURCES) {
-    if (num_resources != 0) {
+  int num_resources = 0;
+  while (true) {
+    OC_PRINTF("\nEnter number of resources in this ACE: ");
+    if (scanf("%d", &num_resources) <= 0) {
```

Comment:
Is it correct to not use `SCANF` macro ?

Reply:
Yes, it's fine, `SCANF` is just a local macro defined in this file that logs error and continues execution:

```C
#define SCANF(...)                                                             \
  do {                                                                         \
    if (scanf(__VA_ARGS__) <= 0) {                                             \
      OC_PRINTF("ERROR Invalid input\n");                                      \
      fflush(stdin);                                                           \
    }                                                                          \
  } while (0)
```